From a473eaa33b9c663b862f3346ba528f22c07f316e Mon Sep 17 00:00:00 2001
Message-Id: <a473eaa33b9c663b862f3346ba528f22c07f316e.1413837857.git.chang-joon.lee@intel.com>
In-Reply-To: <006497b7323eca32c2f489a1a9f662743c0d16d2.1413837855.git.chang-joon.lee@intel.com>
References: <006497b7323eca32c2f489a1a9f662743c0d16d2.1413837855.git.chang-joon.lee@intel.com>
From: Gajanan Bhat <gajanan.bhat@intel.com>
Date: Wed, 23 Jul 2014 19:39:52 +0530
Subject: [PATCH 208/312] FOR_UPSTREAM [VPG]: drm/i915: Cleanup CHV display
 initialization

Strap bits are not working consistently with all BIOS versions. Disable this
check for display initialization till it works with all new BIOS relases.
Remove CHV display initialization within VLV condition.
Remove Nested IS_CHERRYVIEW Checks.

Change-Id: I0b65aafe86fdd195b4167febb95ec841dd88ef17
Signed-off-by: Gajanan Bhat <gajanan.bhat@intel.com>

Conflicts:
	drivers/gpu/drm/i915/intel_display.c
---
 drivers/gpu/drm/i915/intel_display.c |   39 ++++++++++++++++------------------
 1 file changed, 18 insertions(+), 21 deletions(-)

diff --git a/drivers/gpu/drm/i915/intel_display.c b/drivers/gpu/drm/i915/intel_display.c
index 7693e99..a2c0761 100644
--- a/drivers/gpu/drm/i915/intel_display.c
+++ b/drivers/gpu/drm/i915/intel_display.c
@@ -12387,30 +12387,27 @@ static void intel_setup_outputs(struct drm_device *dev)
 
 		if (I915_READ(PCH_DP_D) & DP_DETECTED)
 			intel_dp_init(dev, PCH_DP_D, PORT_D);
+	} else if (IS_CHERRYVIEW(dev)) {
+
+		/* Strap bits not working consistently with all BIOS versions.
+		 * For now initialize all displays and enable them only if
+		 * actual display is present.
+		 */
+
+		intel_dp_init(dev, VLV_DISPLAY_BASE + DP_C, PORT_C);
+
+		intel_hdmi_init(dev, VLV_DISPLAY_BASE + CHV_HDMID, PORT_D);
+
+		if (dev_priv->vbt.has_mipi)
+			intel_dsi_init(dev);
 	} else if (IS_VALLEYVIEW(dev)) {
 		/* There is no detection method for MIPI so rely on VBT */
-		if (!IS_CHERRYVIEW(dev)) {
-			if (dev_priv->vbt.has_mipi)
-				intel_dsi_init(dev);
-			else if (I915_READ(VLV_DISPLAY_BASE + DP_C) &
-					DP_DETECTED)
-				intel_dp_init(dev, VLV_DISPLAY_BASE + DP_C, PORT_C);
-			if (I915_READ(VLV_DISPLAY_BASE + GEN4_HDMIB) &
-					SDVO_DETECTED) {
-				intel_hdmi_init(dev,
-					VLV_DISPLAY_BASE + GEN4_HDMIB,
-					PORT_B);
-			}
-		} else if (IS_CHERRYVIEW(dev)) {
-			if (I915_READ(VLV_DISPLAY_BASE + CHV_HDMID) &
-					SDVO_DETECTED) {
-				intel_hdmi_init(dev, VLV_DISPLAY_BASE + CHV_HDMID,
-						PORT_D);
-				if (I915_READ(VLV_DISPLAY_BASE + DP_D) &
-						DP_DETECTED)
-					intel_dp_init(dev, VLV_DISPLAY_BASE + DP_D, PORT_D);
-			}
+		if (dev_priv->vbt.has_mipi)
 			intel_dsi_init(dev);
+		else if (I915_READ(VLV_DISPLAY_BASE + DP_C) & DP_DETECTED)
+			intel_dp_init(dev, VLV_DISPLAY_BASE + DP_C, PORT_C);
+		if (I915_READ(VLV_DISPLAY_BASE + GEN4_HDMIB) & SDVO_DETECTED) {
+			intel_hdmi_init(dev, VLV_DISPLAY_BASE + GEN4_HDMIB, PORT_B);
 		}
 	} else if (SUPPORTS_DIGITAL_OUTPUTS(dev)) {
 		bool found = false;
-- 
1.7.9.5

